mobility_games/games/wall_audio_player.py

```diff
@@ -60,22 +63,27 @@ def config_callback(self, config, level):
         return config
 
     def get_dist(self, msg):
-        self.dist = msg.data
-        print(self.dist)
-        if self.speed:
-            filtereddist =  max(min(3*self.dist, 20), 2)
-            self.rate = 1.0/filtereddist;
-            self.r = rospy.Rate(1.0/self.rate)
+        """
+        Code to run when distance is updated on the smart_wall_dist node.
+        """
+        self.dist = msg.data #get distance
+        print(self.dist) #print distance
+        if self.speed: #if modulating speed
+            filtereddist =  max(min(3*self.dist, 20), 2) #modulate based on this function (CHANGE THIS FUNCTION EVENTUALLY TO BE BASED ON MAXFREQDIST AND ALTSOUNDDIST)
```

Comment:
Seems like you have a some function based on maxfreqdist and altsounddist

```
 multiplier = self.maxfreq/math.pow(2, (self.maxfreqdist-self.altsounddist)) #calculate multiplier based on maxfreq, maxfreqdist, and altsounddist
 freq = max(min(multiplier*math.pow((abs(self.dist+random.random()*self.randrange)-self.altsounddist), 2)+65.4*2, self.maxfreq), 65.4*2)*2 #set frequency based on distance
```

do these have a relation or am I mistaken?